### PR TITLE
More jclouds location unit tests

### PR DESCRIPTION
**Pull request by aledsage**
Following on from @bostko's great changes in https://github.com/apache/brooklyn-server/pull/366 (to allow us to write more real unit tests, for JcloudsLocation testing), this PR converts the easiest tests into real unit tests.

@duncangrant you may well find this helpful for the JcloudsLocation customizer changes that you are making.

### CODE REVIEW

**Comment by aledsage** [comment]
@bostko or @duncangrant could you take a quick look please - I'd like to get this merged to begin improving the unit test coverage of our `JcloudsLocation` code!

**Comment by bostko** [comment]
Great @aledsage !
`JcloudsLocation` is now better and a lot more tests.

**Comment by aledsage** [comment]
Thanks @bostko - jenkins has confirmed tests pass; merging now.

File: locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsSshMachineLocation.java [hunk 1]
```diff
@@ -132,8 +132,6 @@
 
     private transient Optional<Image> _image;
 
-    private RunScriptOnNode.Factory runScriptFactory;
```
Comment (neykov) [inline locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsSshMachineLocation.java]: Is this safe for rebind?

**Pull request merged**